src/components/Spotlight/Spotlight.tsx

```diff
@@ -275,32 +285,42 @@ const Spotlight = ({
   ]);
 
   // TODO: use a resize observer to detect when the bounds of the target change
-  const updateWindowBounds = () => {
+  const updateWindowBounds = useCallback(() => {
     setWindowDimensions({
       width: window.innerWidth,
       height: window.innerHeight,
     });
-  };
+  }, [setWindowDimensions]);
 
-  const updateScrollPosition = (e: Event) => {
+  const updateScrollPosition = useCallback(() => {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore documentElement does exist on target element
-    setScrollTop(e?.target?.documentElement.scrollTop);
-  };
+    // setScrollTop(e?.target?.documentElement.scrollTop);
+    setScrollTop(window.scrollY);
```

Comment:
https://www.npmjs.com/package/@react-hook/window-scroll

similar comment on using this for scroll position